var/spack/repos/builtin/packages/py-mypy/package.py

```diff
@@ -13,12 +13,14 @@ class PyMypy(PythonPackage):
     pypi = "mypy/mypy-0.740.tar.gz"
 
     version('0.910', sha256='704098302473cb31a218f1775a873b376b30b4c18229421e9e9dc8916fd16150')
+    version('0.900', sha256='65c78570329c54fb40f956f7645e2359af5da9d8c54baa44f461cdc7f4984108')
     version('0.800', sha256='e0202e37756ed09daf4b0ba64ad2c245d357659e014c3f51d8cd0681ba66940a')
     version('0.740', sha256='48c8bc99380575deb39f5d3400ebb6a8a1cb5cc669bbba4d3bb30f904e0a0e7d')
 
     variant('python2', default=False, description='Enable checking of python 2 type annotations')
 
-    depends_on('python@3.5:', type=('build', 'run'))
+    depends_on("python@3.5:")
+    depends_on("python@:3.8", when="0.900:")
```

Comment:
The error is that you forgot an `@` here, but the error message was not great!

Reply:
ahhhhhhh I didn't even see that! I was struck by bug blindness. :bug: :dark_sunglasses: